akvo/rsr/spa/app/modules/editor/project-init-handler.js

```diff
@@ -64,26 +66,30 @@ const ProjectInitHandler = connect(({editorRdr}) => ({ editorRdr }), actions)(Re
     }
     else resolve()
   })
-  let nextSectionIndex = 0
-  const fetchNextSection = () => {
-    fetchSection(sectionIndexPipeline[nextSectionIndex])
-    .then(() => {
-      props.setSectionFetched(sectionIndexPipeline[nextSectionIndex])
-      if (nextSectionIndex < 10) {
-        nextSectionIndex += 1
-        fetchNextSection()
-      }
-    })
-  }
+
   useEffect(() => {
-    if (prevParams && prevParams.id !== params.id && params.id !== 'new'){
-      fetchSection(3)
-      fetchSection(5)
+    // eslint-disable-next-line no-restricted-globals
+    if (!isNaN(params?.id) && (nextSectionIndex < 11)) {
+      if (!props?.editorRdr[`section${nextSectionIndex + 1}`]?.isFetched) {
```

Comment:
Looks like these can be put into one `if`.

--------------

Also, maybe it's my programming style, but I think something like this looks better

```js
if(badCondition){
  return
}
// initialize variables for next condition
if(otherBadCondition){
  return
}
// actual body of the function
```

Results in less indentation and clear rules with no nested `if`s for the reader to figure out when the function should be doing something.

-----------

Finally, it would be great to use a constant here instead of just `11` (e.g `SECTION_COUNT`). Probably it could be used elsewhere too (not for this PR).

Reply:
yes yes, I'm in. Okay @MichaelAkvo I'm working to fix all feedbacks